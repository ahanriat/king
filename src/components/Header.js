import React from 'react';
import ReactNative from 'react-native';

const {
  StyleSheet,
  View,
  TouchableOpacity,
  Image,
  Text,
} = ReactNative;

const {
  PropTypes,
  Component,
} = React;

const styles = StyleSheet.create({
  container: {
    height: 70,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },

  button: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  icon: {
    width: 20,
    height: 20,
    margin: 25,
  },
  title: {
    flex: 4,
    fontFamily: 'Montserrat-Black',
    fontSize: 16,
    justifyContent: 'center',
    textAlign: 'center',
    color: '#FFFFFF',
  },
  titleBig: {
    fontSize: 26,
  },
});

class Header extends Component {
  render() {
    const {
      onRightButtonPress,
      onLeftButtonPress,
      buttonLeftImage,
      buttonRightImage,
      big,
    } = this.props;

    const styleTitle = [styles.title];
    if (big) {
      styleTitle.push(styles.titleBig);
    }

    return (
      <View style={styles.container}>
        <TouchableOpacity
          style={styles.button}
          onPress={(onLeftButtonPress)}
        >
          <Image
            style={styles.icon}
            source={buttonLeftImage}
          />
        </TouchableOpacity>
        <Text style={styleTitle}>{this.props.title}</Text>
        <TouchableOpacity
          style={styles.button}
          onPress={(onRightButtonPress)}
        >
          <Image
            style={styles.icon}
            source={buttonRightImage}
          />
        </TouchableOpacity>
      </View>
    );
  }

}

Header.propTypes = {
  big: PropTypes.bool,
  title: PropTypes.string,
  onLeftButtonPress: PropTypes.func,
  onRightButtonPress: PropTypes.func,
  buttonLeftImage: PropTypes.number,
  buttonRightImage: PropTypes.number,
};

module.exports = Header;
