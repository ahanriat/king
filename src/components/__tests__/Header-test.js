import React from 'react';
import renderer from 'react-test-renderer'; // eslint-disable-line

import CloseIcon from '../../assets/images/ic_close.png';
import AddIcon from '../../assets/images/ic_add.png';

import Header from '../Header';

describe('Header', () => {
  it('render small', () => {
    const props = {
      title: 'Title Small',
      buttonLeftImage: CloseIcon,
      buttonRightImage: AddIcon,
    };
    const tree = renderer.create(
      <Header {...props} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('render big', () => {
    const props = {
      title: 'Title Big',
      buttonLeftImage: CloseIcon,
      buttonRightImage: AddIcon,
    };
    const tree = renderer.create(
      <Header {...props} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
