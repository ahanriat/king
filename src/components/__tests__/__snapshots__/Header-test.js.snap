exports[`Header render big 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center"
    }
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function bound touchableHandleResponderGrant]}
    onResponderMove={[Function bound touchableHandleResponderMove]}
    onResponderRelease={[Function bound touchableHandleResponderRelease]}
    onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
    onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
    onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1
      }
    }
    testID={undefined}>
    <Image
      source={1}
      style={undefined} />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "flex": 4,
          "fontFamily": "Montserrat-Black",
          "fontSize": 16,
          "justifyContent": "center",
          "textAlign": "center"
        }
      ]
    }>
    Title Big
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function bound touchableHandleResponderGrant]}
    onResponderMove={[Function bound touchableHandleResponderMove]}
    onResponderRelease={[Function bound touchableHandleResponderRelease]}
    onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
    onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
    onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1
      }
    }
    testID={undefined}>
    <Image
      source={1}
      style={undefined} />
  </View>
</View>
`;

exports[`Header render small 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center"
    }
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function bound touchableHandleResponderGrant]}
    onResponderMove={[Function bound touchableHandleResponderMove]}
    onResponderRelease={[Function bound touchableHandleResponderRelease]}
    onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
    onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
    onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1
      }
    }
    testID={undefined}>
    <Image
      source={1}
      style={undefined} />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "flex": 4,
          "fontFamily": "Montserrat-Black",
          "fontSize": 16,
          "justifyContent": "center",
          "textAlign": "center"
        }
      ]
    }>
    Title Small
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function bound touchableHandleResponderGrant]}
    onResponderMove={[Function bound touchableHandleResponderMove]}
    onResponderRelease={[Function bound touchableHandleResponderRelease]}
    onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
    onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
    onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1
      }
    }
    testID={undefined}>
    <Image
      source={1}
      style={undefined} />
  </View>
</View>
`;
